### PR TITLE
Add application context util for SHOGun-Boot

### PR DESCRIPTION
**Pull request by dnlkoch**
:boom: **BREAKING CHANGE** :boom: 

This adds (basic) support for the application context delivered by a shogun-boot backend while integrating a specific app context util.

As the `AppContextUtil` has been renamed and the static access has been removed, projects should now get/import the util as follows:

```ts
import { getAppContextUtil } from './util/getAppContextUtil';

const appContextUtil = getAppContextUtil();
appContextUtil.appContextToState(appContext);
```

Please review @terrestris/devs.

### CODE REVIEW

File: src/config/config.js [hunk 1]
```diff
@@ -14,25 +14,30 @@ if (nodeEnv && nodeEnv.indexOf('production') > -1) {
...
 
 let appContextPath;
+let layerPath;
 if (appMode === 'start:shogun2') {
   appContextPath = shogun2Path;
+  layerPath = basePath + 'rest/layers';
 }
 if (appMode === 'start:static') {
   appContextPath = staticPath;
+  layerPath = basePath + 'rest/layers';
```
Comment (weskamm) [inline src/config/config.js:24]: static mode does not need a `layerPath` as it does not talk to a backend

File: src/config/config.js [hunk 2]
```diff
@@ -14,25 +14,30 @@ if (nodeEnv && nodeEnv.indexOf('production') > -1) {
...
+  layerPath,
+  locale: localePath,
   getBasePath: function (){
     return basePath;
   },
   printAction: `${basePath}print/print`,
   printCreateUrlAction: `${basePath}print/createUrl.action`,
   printUrlAction: `${basePath}print/doPrint.action`,
   printGetResultAction: `${basePath}print/getPrintResult.action`
+
```
Comment (weskamm) [inline src/config/config.js:42]: could be removed

File: src/model/BaseEntity.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import _ from 'lodash';
```
Comment (weskamm) [inline src/model/BaseEntity.ts:1]: i think this leads to a fully included lodash in production builds if i remember state of treeshaking correctly.
would be better to have an explicit import.

File: src/config/config.js (hunk 2 above)
Comment (dnlkoch) [reply src/config/config.js]: Fixed in db57331.

File: src/model/BaseEntity.ts (hunk 3 above)
Comment (dnlkoch) [reply src/model/BaseEntity.ts]: Yes, you're right. Fixed in in fcd272b.

**Comment by weskamm** [comment]
I see breaking changes in renaming the appcontext util. Maybe you can workaround that so that existing applications wont break?

File: src/config/config.js (hunk 1 above)
Comment (dnlkoch) [reply src/config/config.js]: Fixed in 44c1024.

**Review by weskamm — APPROVED** [review]

**Review by annarieger — APPROVED** [review]

**Comment by dnlkoch** [comment]
Thanks for the reviews!

**Pull request merged**